src/utils/connectedResourceDeleter.ts

```diff
@@ -0,0 +1,55 @@
+import { createSelector } from "reselect";
+
+import ApiSlice, { ApiDataStore, PendingErrorState, ResourceType } from "@/store/apiSlice";
+
+export const resourcesDeletedSelector = (resource: ResourceType) => (store: ApiDataStore) =>
+  store.meta.deleted[resource];
+
+/**
+ * Creates a resource deletion async function for the given resource type.
+ *
+ * Resource deletion presents an interesting problem in the Connection system: often when a resource
+ * gets deleted, the connections that depend on it will refetch their resources, or unmount the
+ * child component that depended on that resource. In that case, waiting for redux state changes
+ * within that child component to give feedback when the deletion is successful becomes a race
+ * condition and will often fail. Therefore, for deletions, we have this utility for handling
+ * deletes more gracefully.
+ */
+export const connectedResourceDeleter = (
```

Comment:
Reviewers please read over the comments in this file; it forms the meat of how I'm supporting deletes in v3.